src/Oro/Bundle/FilterBundle/Filter/NumberFilter.php

```diff
@@ -81,9 +81,43 @@ public function getMetadata()
     {
         $metadata = parent::getMetadata();
 
-        $formView = $this->getForm()->createView();
-        $metadata['formatterOptions'] = $formView->vars['formatter_options'];
+        $formBuilder = $this->getFormBuilder();
+        $dataType = $formBuilder->getOption('data_type');
+        $metadata['formatterOptions'] = $this->getFormatterOptions($dataType);
 
         return $metadata;
     }
+
+    /**
+     * TODO: Move this in another service?
+     *
+     * @param string $dataType
+     *
+     * @return array
+     */
+    public function getFormatterOptions($dataType)
+    {
+        $formatterOptions = [];
+
+        switch ($dataType) {
+            case NumberFilterType::DATA_DECIMAL:
+                $formatterOptions['decimals'] = 2;
+                $formatterOptions['grouping'] = true;
+                break;
+            case NumberFilterType::DATA_INTEGER:
```

Comment:
explain it in the signature of the method, and explain why it has been copy/pasted